### ISSUE TITLE
Front end sign in not using the new API endpoint

### ISSUE THREAD

**Issue by aunefyren** [aunefyren/poenskelisten#18]
> 
> Based on your provided information, it seems your front end is not using the newest code. Could it be cached in the browser perhaps? You could try to delete cookies/browser history or do a force-refresh (CTRL + F5).
> 
> The `/api/open/token/register` endpoint is from a previous version.

Thanks for the response. It appears that my issue is unrelated to that of the original author, in that case - would you prefer if I open a new issue for it? Clearing cache etc. wasn't effective - I received the same error after completely removing the Docker image and pulling it again, and attempting to log in using a browser I've never previously used.

_Originally posted by @hyperbrentilate in https://github.com/aunefyren/poenskelisten/issues/17#issuecomment-1870281021_

**Comment by aunefyren** [comment]
Are you able to confirm what version Pønskelisten is running on within the `config.json`? Also, are you using Cloudflare in front of your site?

**Comment by hyperbrentilate** [comment]
The version in my config.json is v2.0.3.

I have all my services reverse proxied via linuxserver.io's [SWAG](https://github.com/linuxserver/docker-swag) container. I was coincidentally updating some of my proxy settings the other day as well, which could be related.

I do use Cloudflare, but only for DNS; I have their proxying off.

To continue testing I've reset my database and attempted to re-register an admin user. My Docker log now shows a POST to `/api` rather than `/api/open/token/register` which, I assume, is the expected behavior. I still receive the error about not being able to reach the API, though,

```
POST (domain)/api/open/users 404 (Not Found) functions.js:8
send_registration @ register.js:118
onsubmit @ register:121
SyntaxError: Unexpected non-whitespace character after JSON at position 4 (line 1 column 5) - Response: 404 page not found register.js:92 
```
However, this time I'm not even able to register, whereas I was able to before. As a result, it sounds like something might be breaking in the reverse proxying. The front page loads fine though, and none of my other services have been impacted by my changes at all, so it's challenging to identify what specifically is causing this.

I understand if this is moving outside the scope of what you're interested in supporting, but I'm happy to try whatever troubleshooting steps you would like if you think it'll be useful feedback. I'll also explore walking back my proxy changes to try to identify what the breaking change could've been. Clearly I should've known better than to update settings that were working, whoops.

**Comment by aunefyren** [comment]
Interesting. No worries about what to support, I am happy to help if I can. Either something is broken, or could be documented and explained better.

The v2.0.0 changed some API endpoints, and made the URL structure more consistent. This includes that object types in the API URL are all plural. `/token/` was changed to `/tokens/` for example. So loading the front page is all the same, but the JavaScript in the front end is attempting to contact the old API URL, suggesting the code is old.

I have had issues removing cached code in the front end when using Cloudflare, I typically have to "purge the cache" in the dashboard, but I would assume the proxy has to be enabled?

**Comment by hyperbrentilate** [comment]
Resolved. I figured it out when you mentioned the endpoint structure - I wasn't getting a path with plurals, I was just getting `/api` as the entire path after changing things.

I believe what might've happened is that I was, initially, running into an issue with caching, as you suggested. I didn't realize that at the time though, thinking it might've been an issue with the way the function paths were interacting with the reverse proxying. I experimented with this through forcing different path resolution in my proxy conf (e.g., adding a separate location for `/api` etc.).

After clearing the cache and seeing just `api` show up in the logs, I realized that I'd forgotten to restart the SWAG container to remove that location rewrite. So I think it was a cache issue, I just broke something else in trying to resolve that, which kept things broken after the original issue was fixed,

I'll report back if I discover anything else that seems related to this, but as of now I've been able to create a new user and log in as that user, so everything looks good.

Thanks again for the responses - it doesn't look like there's anything broken or in need of better documentation on your end, this was all me.
